primitives/src/transaction.rs

```diff
@@ -89,3 +93,133 @@ impl Version {
 impl fmt::Display for Version {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result { fmt::Display::fmt(&self.0, f) }
 }
+
+/// A reference to a transaction output.
+///
+/// ### Bitcoin Core References
+///
+/// * [COutPoint definition](https://github.com/bitcoin/bitcoin/blob/345457b542b6a980ccfbc868af0970a6f91d1b82/src/primitives/transaction.h#L26)
+#[derive(Copy, Clone, Debug, Eq, Hash, PartialEq, PartialOrd, Ord)]
+pub struct OutPoint {
+    /// The referenced transaction's txid.
+    pub txid: Txid,
+    /// The index of the referenced output in its transaction's vout.
+    pub vout: u32,
+}
+#[cfg(feature = "serde")]
+internals::serde_struct_human_string_impl!(OutPoint, "an OutPoint", txid, vout);
+
+impl OutPoint {
+    /// The number of bytes that an outpoint contributes to the size of a transaction.
+    pub const SIZE: usize = 32 + 4; // The serialized lengths of txid and vout.
+
+    /// Creates a new [`OutPoint`].
+    #[inline]
+    pub const fn new(txid: Txid, vout: u32) -> OutPoint { OutPoint { txid, vout } }
+
+    /// Creates a "null" `OutPoint`.
+    ///
+    /// This value is used for coinbase transactions because they don't have any previous outputs.
+    #[inline]
+    pub fn null() -> OutPoint { OutPoint { txid: Txid::all_zeros(), vout: u32::MAX } }
+
+    /// Checks if an `OutPoint` is "null".
+    #[inline]
+    pub fn is_null(&self) -> bool { *self == OutPoint::null() }
+}
+
+impl Default for OutPoint {
+    fn default() -> Self { OutPoint::null() }
+}
+
+impl fmt::Display for OutPoint {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}:{}", self.txid, self.vout)
+    }
+}
+
+#[cfg(feature = "alloc")]       // Because of `parse::ParseIntError`.
+impl core::str::FromStr for OutPoint {
```

Comment:
I object to this impl existing unless the format is super-wide standard. I've seen it used at multiple places but I'm not really 100% sure it's sensible.

Reply:
Assuming this is the `txid:vout` format, this is super common and used all over the place. There is also no other string format for outpoints that I have ever seen.

Reply:
I vaguely remember seeing other things but I don't remember what exactly it was. Probably a different separator or no separator. Anyway, it's quite likely that this is fine.

Reply:
The `bitcoin-tx` tool uses this format to take outpoints on the command-line. The format <64 hex>:<number> is also widely recognized as being an outpoint, and appears on slides and on mailing list posts etc.

The format is also parsed by `hal`, the Liquid functionary code, and probably esplora ... though those don't "really" count since they are all using rust-bitcoin.